Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -4311,6 +4320,30 @@ namespace AssetProcessor
         }
     }
 
+    AZStd::string AssetProcessorManager::ComputeRecursiveDependenciesFingerprint(const AZStd::string& fileAbsolutePath, const AZStd::string& fileDatabaseName)
+    {
+        AZStd::string concatenatedFingerprints;
+
+        // QSet is not ordered.
+        SourceFilesForFingerprintingContainer knownDependenciesAbsolutePaths;
+        // this automatically adds the input file to the list:
+        QueryAbsolutePathDependenciesRecursive(QString::fromUtf8(fileDatabaseName.c_str()), knownDependenciesAbsolutePaths, SourceFileDependencyEntry::DEP_Any, false);
+        AddMetadataFilesForFingerprinting(QString::fromUtf8(fileAbsolutePath.c_str()), knownDependenciesAbsolutePaths);
+
+        // reserve 17 chars for each since its a 64 bit hex number, and then one more for the dash inbetween each.
+        concatenatedFingerprints.reserve((knownDependenciesAbsolutePaths.size() * 17));
```

Comment:
Can you make this 17 a const, or better yet calculate it based on the data used? This seems potentially frail to an upstream change. If someone changes the underlying data structure, it might be difficult to trace unexpected side effects back here.